fastq_filtrator.py

```diff
@@ -0,0 +1,182 @@
+# It seems very heavily memory-consuming and tends to crash
+# However, I am unsure how to fix this. Maybe it would be better to just take 4 lines of the file and work with them
+# 1. main function (just combines all the rest). works through iterating over lists
+def main(input_fastq, output_file_prefix, gc_bounds, length_bounds, quality_threshold, save_filtered):
+    # opens
+    open_file(input_fastq, full_length_file)
+    # now we have open the file and created a "full length file" list
+    for read in full_length_file:
+        gc_bounds_function(read, gc_bounds, reads_passed_GC, reads_not_passed_GC)
+    for read in reads_passed_GC:
+        length_bounds_function(read, length_bounds, reads_passed_GC_length, reads_not_passed_GC_length)
+    for read in reads_passed_GC_length:
+        threshold_quality_function(read, reads_passed_GC_length_quality)
+    filename_passed = str(output_file_prefix) + "_passed.fastq"
+    filename_failed = str(output_file_prefix) + "_failed.fastq"
+    writing_results(filename_passed, filename_failed)
+
+
+# In[79]:
+
+
+# OPENING THE FILE
+def divide_into_read_parts(lines=None):
+    name_parts = ['name', 'sequence', 'optional', 'quality']
+    return {k: v for k, v in zip(name_parts, lines)}
+    # maybe it will be better not to create it later so it does not take space
+    # the idea was found at https://www.biostars.org/p/317524/
+
+
+def open_file(input_fastq, full_length_file):
+    with open(input_fastq, 'r') as f:
+        lines = []
+        for line in f:
+            lines.append(line.rstrip())
+            if len(lines) == 4:
+                record = divide_into_read_parts(lines)
+                full_length_file.append(record)
+                lines = []
+
+
+# In[80]:
+
+
+# GC COUNT RELATED FUNCTIONS
+def GC_count(read):
+    # sub-function 1 for gc bounds one
+    # just the sum of C and G occurrences over the length of this read
+    return 100*(read['sequence'].count("C")+read['sequence'].count("G"))/len(read['sequence'])
+
+
+# In[81]:
+
+
+# If it will be the first function, it would take the initial list and after it
+# the reads that pass further are saved in reads_passed_GC
+def gc_bounds_function(read, gc_bounds, reads_passed_GC, reads_not_passed_GC):
+    gc_bounds_number = list()
+    gc_bounds = gc_bounds.split()
+    if len(gc_bounds) == 1:
+        if gc_bounds[0] == "default":
+            gc_bounds_number = (0, 100)
+        elif gc_bounds[0].isdigit():
+            gc_bounds_number = (0, int(gc_bounds[0]))
+    elif len(gc_bounds) == 2 and gc_bounds[0].isdigit() and gc_bounds[1].isdigit():
+        if int(gc_bounds[0]) <= int(gc_bounds[1]):
+            gc_bounds_number = (int(gc_bounds[0]), int(gc_bounds[1]))
+    if GC_count(read) >= gc_bounds_number[0] and GC_count(read) <= gc_bounds_number[1]:
+        reads_passed_GC.append(read)
+    else:
+        if save_filtered.lower() == "true":
```

Comment:
Логичнее было бы ожидать, что save_filtered имеет логический тип
```suggestion
        if save_filtered.lower() == True":
```